sandbox/samples/cpp/alembicassembly/alembicassembly.cpp

```diff
@@ -0,0 +1,1525 @@
+            // generate xform sequence stack from our camera
+            const std::vector<asr::TransformSequence> xform_seq_stack = xform_seq_for_obj(
+                cam_obj,
+                shutter_open,
+                shutter_close,
+                time_offset);
+
+            if (xform_seq_stack.size())
+            {
+                const asr::TransformSequence xform_seq = flatten_xform_seq(xform_seq_stack);
+
+                m_camera->transform_sequence() = xform_seq;
+            }
+
+            // of course, we don't forget to call the camera method
+            return m_camera->on_render_begin(project, abort_switch);
+        }
+
+        void spawn_ray(
+            asr::SamplingContext&    sampling_context,
+            const asf::Dual2d&       ndc,
+            asr::ShadingRay&         ray) const override
+        {
+            m_camera->spawn_ray(sampling_context, ndc, ray);
+        }
+
+        bool connect_vertex(
+            asr::SamplingContext&    sampling_context,
+            const float              time,
+            const asf::Vector3d&     point,
+            asf::Vector2d&           ndc,
+            asf::Vector3d&           outgoing,
+            float&                   importance) const override
+        {
+            m_camera->connect_vertex(sampling_context, time, point, ndc,
+                                     outgoing, importance);
+        }
+
+        bool project_camera_space_point(
+            const asf::Vector3d&     point,
+            asf::Vector2d&           ndc) const override
+        {
+            m_camera->project_camera_space_point(point, ndc);
+        }
+
+        bool project_segment(
+            const float              time,
+            const asf::Vector3d&     a,
+            const asf::Vector3d&     b,
+            asf::Vector2d&           a_ndc,
+            asf::Vector2d&           b_ndc) const override
+        {
+            m_camera->project_segment(time, a, b, a_ndc, b_ndc);
+        }
+
+      private:
+        // Parameters.
+        std::string m_file_path;
+        std::string m_obj_path;
+
+        const char* Model = "alembic_camera";
+
+        asf::auto_release_ptr<asr::Camera> m_camera;
+    };
+
+
+    //
+    // Alembic camera factory.
+    //
+
+    class AlembicCameraFactory
+      : public asr::ICameraFactory
+    {
+      public:
+        // Delete this instance.
+        void release() override
+        {
+            delete this;
+        }
+
+        // Return a string identifying this camera model.
+        const char* get_model() const override
+        {
+            return Model;
+        }
+
+        // Return metadata for this camera model.
+        asf::Dictionary get_model_metadata() const override
+        {
+            return
+                asf::Dictionary()
+                    .insert("name", Model)
+                    .insert("label", "Alembic Camera")
+                    .insert("default_model", "true");
```

Comment:
The default camera model is the pinhole camera; remove `default_model` altogether.

Reply:
Done.